rust-runtime/aws-smithy-eventstream/fuzz/fuzz_targets/round_trip.rs

```diff
@@ -4,25 +4,16 @@
  */
 
 #![no_main]
-use aws_smithy_eventstream::error::Error;
-use aws_smithy_eventstream::frame::Message;
+use aws_smithy_eventstream::arbitrary::ArbMessage;
+use aws_smithy_eventstream::frame::{read_message_from, write_message_to};
 use libfuzzer_sys::fuzz_target;
 
-fuzz_target!(|message: Message| {
+fuzz_target!(|message: ArbMessage| {
+    let message = message.into();
     let mut buffer = Vec::new();
-    match message.write_to(&mut buffer) {
-        Err(
-            Error::HeadersTooLong
-            | Error::PayloadTooLong
-            | Error::MessageTooLong
-            | Error::InvalidHeaderNameLength
-            | Error::TimestampValueTooLarge(_),
-        ) => {}
-        Err(err) => panic!("unexpected error on write: {}", err),
-        Ok(_) => {
-            let mut data = &buffer[..];
-            let parsed = Message::read_from(&mut data).unwrap();
-            assert_eq!(message, parsed);
-        }
+    if write_message_to(&message, &mut buffer).is_ok() {
+        let mut data = &buffer[..];
+        let parsed = read_message_from(&mut data).unwrap();
+        assert_eq!(message, parsed);
     }
```

Comment:
Is this intentional that we no longer `panic!` on an error?

Reply:
It should've kept `panic!` on an error. See https://github.com/awslabs/smithy-rs/pull/3139#discussion_r1380583526.